### ISSUE TITLE
MacOS toolchain potentially broken

### ISSUE THREAD

**Issue by Photonios** [minoca/os#126]
I tried to build MinocaOS on the following system today:

<img width="589" alt="screen shot 2017-07-01 at 14 44 07" src="https://user-images.githubusercontent.com/5777517/27761775-b02aab4e-5e6b-11e7-8baa-c922da56a119.png">

<img width="531" alt="screen shot 2017-07-01 at 14 44 59" src="https://user-images.githubusercontent.com/5777517/27761783-cbf537b8-5e6b-11e7-99fa-7ff634921aad.png">

I ran into the following problems:

* Various errors during the build caused by warnings that were turned into errors by `-Werror`. I managed to get past those using `EXTRA_CFLAGS=-Wno-error`.

* Comments (denoted by `##`) in assembler file were somehow seen as syntax errors by Clang. I managed to work around that by using a simple script that removes all the comments:

    ```
    find . -name \*.inc -o -name \*.S | while read -d $'\n' file
    do
        sed -i.bak '/^##/d' $file
        rm $file.bak
     done
     ```

     I think this is being caused by: https://bugs.llvm.org/show_bug.cgi?id=28921
     but I haven't been able to confirm this.

* During the build it tries to invoke `chalk`, the embedded scripting language that comes with MinocaOS from what I understood. Either it's not properly added to the path during the build, or I don't know. In the end, adding `$SRCROOT/x86dbg/tools/bin` to my `$PATH` fixed it.

TL;DR: Building was a bit shaky and required some work-arounds. Results seems to run fine in qemu.

**Comment by evangreen** [comment]
Nice sedmanship. Glad you got things working eventually, sorry to hear it was so much trouble.

The PATH issue is mentioned in the Environment section of the Minoca README. Perhaps that's a little buried. There's also a handy setenv.sh in the root of the project that you can source into your environment to save yourself that step. That isn't in the readme, and probably should be.

I'm on XCode 8.0 (8A218a). Let me try upgrading XCode and see if I can reproduce the issues you were having...

**Comment by evangreen** [comment]
Pushed b3d700b86ebace8ee9e2d373e14a0e9ab44672c3 and one other change. Give the latest master a try.

**Comment by Photonios** [comment]
Ah, I misread the thing about the `PATH`. My apologies. Seems to work fine now out of the box :) Thanks for the fix!

When I have some spare time, I'll try to contribute some changes. You can close the issue :)

**Comment by evangreen** [comment]
Excellent! Looking forward to it.
